File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m = len(mat)
        n = len(mat[0])

        rows = [0] * m
        cols = [0] * n

        for i in range(m):
            for j in range(n):
                if mat[i][j] == 1:
                    rows[i] += 1
                    cols[j] += 1

        ans = 0
        for i in range(m):
            for j in range(n):
                if mat[i][j] == 1 and rows[i] == 1 and cols[j] == 1:
                    ans += 1

        return ans

        
        

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {

        int[] rows = new int[mat.length];
        int[] cols = new int[mat[0].length];

        for (int i = 0; i< rows.length; i++) {
            for (int j = 0; j<cols.length; j++) {
                int ele = mat[i][j];
                if (ele == 1) {
                    rows[i] += 1;
                    cols[j] += 1;
                } 
            }
        }

        int ans = 0;
        for (int i = 0; i<rows.length; i++) {
            for (int j = 0; j<cols.length; j++) {
                if (mat[i][j] == 1 && rows[i] == 1 && cols[j] == 1) {
                    ans += 1;
                }
            }
        }

        return ans;
        
    }
}